.gitignore

```diff
@@ -1,5 +1,11 @@
 .benchmarks.done
 .browsers
+__pycache__
 PUBLIC
 PRIVATE
-demos
\ No newline at end of file
+add_demo_src/__pycache__
+add_demo_src/tests/__pycache__
+demos
+results
+demo_list.json
```

Comment:
This PR adds a file `demo_list.json`, but then places this file in .gitignore? I wonder if this is done due to the mocked unit test? If so, maybe the whole test working directory where modifications are done could instead be added to .gitignore (rather than singling out individual test files that are modified)?

Reply:
I put this in because I was assuming that we wouldn't want to push local updates to the `demo_list.json` file

Reply:
Oh I see. I wonder if we could then not add the demo_list.json file in the repo at all? Adding a file in the repo, but then simultaneously placing it in .gitignore is something of an untypical mechanism, so would be nice to avoid that kind of pattern.

Reply:
Gotcha. Would you recommend instead searching for the file in the user's file system and then creating it if it doesn't exist?